contracts/exchange/api/RiskCheck.sol

```diff
@@ -34,9 +134,7 @@ contract RiskCheck is BaseContract, MarginConfigContract {
 
     if (isLiquidation && beforeTrade) {
       require(!isAboveMaintenanceMargin(sub), "subaccount liquidated is above maintenance margin");
-    } else if (isLiquidation && !beforeTrade) {
-      _requireNonNegativeValue(sub);
-    } else {
+    } else if (!isLiquidation && !beforeTrade) {
```

Comment:
this seems pretty tricky to detect

Reply:
it was caught by one of the BDD test case, but otherwise it is hard to detect. Risk code [here](https://github.com/gravity-technologies/platform/blob/f4ff35d65d8789be68a12ec9c0b9f755dd229dad/backend/svc/risk/pkg/risk/match_execute.go#L87).